### PR TITLE
Adding queries for collecting QDS data in SQL MI

### PR DESCRIPTION
**Pull request by masree**
### Required for all PRs:

- [ ] Updated associated README.md.
- [ ] Wrote appropriate unit tests.

Summary of PR:

Igor started the [PR](https://github.com/influxdata/telegraf/pull/8465/files) to collect data from query store runtime and wait statistics for SQL DB and SQL MI. Continuing his PR to completion as he does not own this feature anymore. There are a few review comments that needed to be addressed from old PR. Steps to address them:
1. Split the PR into 2 PR - SQL DB and SQL MI
2. Address the review comments related to SQL MI
3. Get it reviewed from community
4. Merge SQL MI PR
5. Create a new PR for SQL DB
6. Repeat 2-4

### CODE REVIEW

File: plugins/inputs/sqlserver/sqlserver.go [hunk 1]
```diff
@@ -317,10 +332,17 @@ func (s *SQLServer) gatherServer(pool *sql.DB, query Query, acc telegraf.Accumul
 			return err
 		}
 	}
+
+	if rows.Err() == nil && query.HasCachedData && rows.NextResultSet() && rows.Next() {
+		var newCachedData string
+		rows.Scan(&newCachedData)
```
Comment (Jagularr) [inline plugins/inputs/sqlserver/sqlserver.go:338]: need to check error

**Review by Jagularr — COMMENTED** [review]
Makes sense to me

**Comment by masree** [comment]
Closing this PR. On second thought, I think it makes sense to send PR for SQL DB first followed by SQL MI. SQL MI is little more complicated than SQL DB as we need to iterate through individual databases inside SQL MI. Here is the link to new PR for SQL DB - #9150

File: plugins/inputs/sqlserver/sqlserver.go (hunk 1 above)
Comment (masree) [reply plugins/inputs/sqlserver/sqlserver.go:338]: Addressed the issue in new PR for SQL DB - #9150

**Pull request closed without merging**